### PR TITLE
DO NOT MERGE - Version 6

### PR DESCRIPTION
**Pull request by yardnsm**
This PR is for the `verson_6`, which contains all of the greatest and latests version 6 changes. 🎉 

All of the changes are documented in https://github.com/wix/wix-style-react/issues/2181. When creating a new PR for this branch, make sure to pick this branch as the base branch.

- [x] All changes merged
- [x] Added migration guide
- [x] Added changes to `CHANGELOG`

### CODE REVIEW

File: docs/migration/v5-v6.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+The component's margins has been updated according to the correct UX guidelines. The `useOldMargins`
+prop has been removed and the new margins were applied by default.
+
+### `<Text/>`
+
+The deprecated `bold` prop has been removed in Version 6. You should use the newer `weight` prop,
+like so:
+
+- `<Text bold="false">` -> `<Heading weight="thin">`
+- `<Text bold="true">` -> `<Heading weight="normal">`
```
Comment (yardnsm) [inline docs/migration/v5-v6.md]: Also, the `./src/Text/MIGRATION.md` stated that `bold={true}` should become `weight=`**`"normal"`**, not `weight=`**`"bold"`**. Am I missing something?

File: docs/migration/v5-v6.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+# Upgrading `wix-style-react` from v5.0.0 to v6.0.0
+
+## Components
+
+### `<MessageBoxFixedHeaderFooter/>`
+
+The component was removed.
```
[the remark this answers is not in the document]
Comment (shlomitc) [reply docs/migration/v5-v6.md]: keep it here, just for a case we missed something

File: docs/migration/v5-v6.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (shlomitc) [reply docs/migration/v5-v6.md]: I like that everything is in one place.
you can add the link to the doc anyway

Comment (shlomitc) [reply docs/migration/v5-v6.md]: I think it's ok. the purpose was to add another level of boldness, so bold became normal

File: docs/migration/v5-v6.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+prop instead.
+
+### `<MessageBoxMarketerialLayout/>`
+
+The `fixImagePosition` prop, which enabled a fixed image position, was removed in Version 6 and it's
+behaviour in now true by default.
+
+### TPA components
+
+All of the TPA component has been removed from the library in Version 6.
```
[the remark this answers is not in the document]
Comment (shlomitc) [reply docs/migration/v5-v6.md]: no need, that's enough

File: docs/migration/v5-v6.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+- For `<TextArea/>`, refer to [_3.2a Text Area_](https://wix-wix-style-react.surge.sh/?selectedKind=3.%20Inputs&selectedStory=3.2a%20Text%20Area&full=0&addons=0&stories=1&panelRight=0) updated documentation.
+- For `<DropdownComposite/>`, refer to [_4.1 Dropdown_](https://wix-wix-style-react.surge.sh/?selectedKind=4.%20Selection&selectedStory=4.1%20Dropdown&full=0&addons=0&stories=1&panelRight=0) updated documentation.
+
+### TPA components
+
+All of the TPA component has been removed from the library in Version 6.
+
+### Backoffice components
+
+The `<TextLink/>` and `<Tooltip/>` **Backoffice** components has been removed from the library. You
```
Comment (shlomitc) [inline docs/migration/v5-v6.md]: `Button` and `ButtonLayout` should also be mentioned here

Comment (yardnsm) [reply docs/migration/v5-v6.md]: Not merged to this branch yet

**Comment by yardnsm** [comment]
Closing in favor of https://github.com/wix/wix-style-react/pull/2833.

**Pull request closed without merging**